woocommerce/woocommerce-ios
# [Order Creation] Prevent merchant from adding parent product to an order via barcode scanning

Closes: #12566


## Description


Scanning a barcode of a parent product previously added it to the order.

This isn’t possible in other ways, as normally a variation should be selected.

The implementation here shows the new order with a notice informing the user of the reason their product isn’t added. This matches Android.

A future improvement might be to open the product variation picker for the relevant product.

## Testing instructions


1. Launch the app, and log in to a store with variable products that have barcodes set for their SKU. Ensure the parent product also has a barcode SKU
2. Tap `Orders`
3. Tap scan
4. Scan the parent product barcode
5. Observe that the order form opens with a notice to say why the parent product wasn't added
6. Retry with a variation barcode
7. Observe that the order form opens with the variation
8. Try with an ordinary product, verify that it's added as expected

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/65c84b69-d5e7-4a6f-9a35-00351b8a8977



---
